lib/src/dag_set_scheduler.rs

```diff
@@ -126,8 +126,21 @@ pub trait DAGSetSchedulerBase<T: ProcessorBase + Clone> {
         self.get_processor_mut()
             .allocate_specific_core(core_i, node);
         let current_time = self.get_current_time();
-        self.get_log_mut()
-            .write_allocating_job(node, core_i, job_i - 1, current_time);
+        if node.params.contains_key("is_preempted") {
+            self.get_log_mut().write_job_event(
+                node,
+                core_i,
+                job_i - 1,
+                JobEventTimes::ResumeTime(current_time),
+            )
+        } else {
+            self.get_log_mut().write_job_event(
+                node,
+                core_i,
+                job_i - 1,
+                JobEventTimes::StartTime(current_time),
+            )
+        }
```

Comment:
This process should be implemented in the XXXLog class.

Reply:
Implementation moved to DAGSetSchedulerLog.